### ISSUE TITLE
docker: fc: Fail on vsock test 

### ISSUE THREAD

**Issue by likebreath** [kata-containers/tests#2398]
# Description of problem

As discussed in issue #2397, we have a regression from the CI on firecracker since #2383 is being fixed. 

# Expected result

The 'vsock_test' should pass with firecracker and vsock is being used.

# Actual result
```
19:09:18 Running command '/usr/bin/docker [docker ps -a -f name=L6Pk7DX6B1Tip2v4ASALIKoDarkZ11 --format {{.Status}}]'
19:09:18 [docker ps -a -f name=L6Pk7DX6B1Tip2v4ASALIKoDarkZ11 --format {{.Status}}]
19:09:18 Timeout: 120 seconds
19:09:18 Exit Code: 0
19:09:18 Stdout: 
19:09:18 Stderr: 
19:09:18 
19:09:18 • Failure [4.899 seconds]
19:09:18 vsock test
19:09:18 /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/vsock_test.go:13
19:09:18   when using vsock
19:09:18   /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/vsock_test.go:31
19:09:18     should print the agent logs in the shim journal [It]
19:09:18     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/vsock_test.go:48
19:09:18 
19:09:18     Expected
19:09:18         <string>: Mar 19 02:09:05 ubuntu1804-azurec22fa0 kata-shim[75476]: time="2020-03-19T02:09:05.834903172Z" level=info msg="copy stdout failed" container=ac18e34889e3b33a20d078ea27964adb906711d8c0d9e7111008718149ebcae8 error="rpc error: code = Unknown desc = EOF" exec-id=ac18e34889e3b33a20d078ea27964adb906711d8c0d9e7111008718149ebcae8 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:05 ubuntu1804-azurec22fa0 kata-shim[75476]: time="2020-03-19T02:09:05.836012081Z" level=info msg="using shim to proxy exit code" container=ac18e34889e3b33a20d078ea27964adb906711d8c0d9e7111008718149ebcae8 exec-id=ac18e34889e3b33a20d078ea27964adb906711d8c0d9e7111008718149ebcae8 exitcode=127 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:10 ubuntu1804-azurec22fa0 kata-shim[75796]: time="2020-03-19T02:09:10.388229044Z" level=info msg=announce agent-socket="hvsock:///run/vc/firecracker/69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831/root/kata.hvsock:1024" container=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 debug=true exec-id=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 log-level=debug name=kata-shim pid=1 proxy-exit-code=true source=shim terminal=false tracing=false version=1.11.0-alpha1-6a828a430c3d35e6ee22ee50a4fd2ed61280ad42
19:09:18         Mar 19 02:09:10 ubuntu1804-azurec22fa0 kata-shim[75796]: time="2020-03-19T02:09:10.8486608Z" level=info msg="copy stdout failed" container=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 error="rpc error: code = Unknown desc = EOF" exec-id=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:10 ubuntu1804-azurec22fa0 kata-shim[75796]: time="2020-03-19T02:09:10.849520307Z" level=info msg="copy stderr failed" container=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 error="rpc error: code = Unknown desc = EOF" exec-id=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:10 ubuntu1804-azurec22fa0 kata-shim[75796]: time="2020-03-19T02:09:10.850624016Z" level=info msg="using shim to proxy exit code" container=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 exec-id=69f28282691fc3242a690641937ff5938a4b97b29f55dc70c7106038e3c29831 exitcode=137 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:15 ubuntu1804-azurec22fa0 kata-shim[76139]: time="2020-03-19T02:09:15.454654641Z" level=info msg=announce agent-socket="hvsock:///run/vc/firecracker/b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3/root/kata.hvsock:1024" container=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 debug=true exec-id=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 log-level=debug name=kata-shim pid=1 proxy-exit-code=true source=shim terminal=false tracing=false version=1.11.0-alpha1-6a828a430c3d35e6ee22ee50a4fd2ed61280ad42
19:09:18         Mar 19 02:09:15 ubuntu1804-azurec22fa0 kata-shim[76139]: time="2020-03-19T02:09:15.678506364Z" level=info msg="copy stdout failed" container=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 error="rpc error: code = Unknown desc = EOF" exec-id=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:15 ubuntu1804-azurec22fa0 kata-shim[76139]: time="2020-03-19T02:09:15.678937768Z" level=info msg="copy stderr failed" container=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 error="rpc error: code = Unknown desc = EOF" exec-id=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 name=kata-shim pid=1 source=shim
19:09:18         Mar 19 02:09:15 ubuntu1804-azurec22fa0 kata-shim[76139]: time="2020-03-19T02:09:15.680299979Z" level=info msg="using shim to proxy exit code" container=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 exec-id=b98d1489abfad6b54284bbd87492101a3a5433f657ce70a282ecfa28af3a84d3 exitcode=0 name=kata-shim pid=1 source=shim
19:09:18         
19:09:18     to contain substring
19:09:18         <string>: source=agent
19:09:18 
19:09:18     /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/vsock_test.go:64
19:09:18 ------------------------------
19:09:18 SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSS
19:09:18 
19:09:18 Summarizing 1 Failure:
19:09:18 
19:09:18 [Fail] vsock test when using vsock [It] should print the agent logs in the shim journal 
19:09:18 /tmp/jenkins/workspace/kata-containers-tests-ubuntu-1804-PR-firecracker/go/src/github.com/kata-containers/tests/integration/docker/vsock_test.go:64
19:09:18 
19:09:18 Ran 55 of 250 Specs in 383.271 seconds
19:09:18 FAIL! -- 54 Passed | 1 Failed | 0 Pending | 195 Skipped
19:09:18 --- FAIL: TestIntegration (383.28s)
19:09:18 FAIL
```

**Comment by chavafg** [comment]
@likebreath any idea if this issue still persist?

**Comment by jodh-intel** [comment]
It would be interesting to know if this fails with the latest version of fc too (0.22.0).

**Comment by likebreath** [comment]
@chavafg Good point. I just tried the fc+kata w/ the `vsock test` and it is now working. I just sent a PR to enable this test in the FC CI: https://github.com/kata-containers/tests/pull/2810.

**Comment by chavafg** [comment]
cool, thank you @likebreath

**Comment by likebreath** [comment]
> It would be interesting to know if this fails with the latest version of fc too (0.22.0).

@jodh-intel That's a good question. I just tried FC v0.22.0, and  it looks like the fc v0.22.0 does not work with kata out of the box. I failed to launch a simple container with fc v0.22.0. Does it work on your side?

Here is I setup: I installed fc v0.22.0 w/ `.ci/install_firecracker.sh` after changing the `firecracker` version to v0.22.0 from `runtime/versions.yaml`. Everything else is Kata 1.x upstream. I run the simple docker workload: `docker run  --runtime=kata-runtime -ti busybox sh`, and got the error: `docker: Error response from daemon: OCI runtime create failed: Failed to connect to firecrackerinstance (timeout 10s): unknown.`
